src/main.test.ts

```diff
@@ -0,0 +1,433 @@
+
+  it('should send success notification if last run failed and current succeed', async () => {
+    await run();
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: success');
+    expect(mockCore.info).toHaveBeenCalledWith('Success notification');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/failure',
+    );
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Message body: {"username":"workflow-status-slack-notification CI alert","icon_emoji":":bangbang:","attachments":[{"color":"good","author_name":"workflowactor","author_link":"https://github.com/workflowactor","author_icon":"https://github.com/workflowactor.png?size=32","fields":[{"title":"Repository","value":"workflow-status-slack-notification","short":true},{"title":"Branch","value":"main","short":true},{"title":"Action URL","value":"<https://github.com/reside-eng/workflow-status-slack-notification/actions/runs/23456|Publish Action>","short":true},{"title":"Event","value":"pull_request","short":true},{"title":"Publish Action workflow success","value":"Previously failing Publish Action workflow in workflow-status-slack-notification succeed.","short":false}]}]}',
+    );
+  });
+
+  it('should not send notification if last run succeeded and current succeed', async () => {
+    await run();
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: success');
+    expect(mockCore.info).toHaveBeenCalledWith('No notification needed');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/success',
+    );
+  });
+
+  it('should send failure notification if last run succeeded and current fails', async () => {
+    mock.inputs['current-status'] = 'failure';
+
+    await run();
+
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: failure');
+    expect(mockCore.info).toHaveBeenCalledWith('Failure notification');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/success',
+    );
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Message body: {"username":"workflow-status-slack-notification CI alert","icon_emoji":":bangbang:","attachments":[{"color":"danger","author_name":"workflowactor","author_link":"https://github.com/workflowactor","author_icon":"https://github.com/workflowactor.png?size=32","fields":[{"title":"Repository","value":"workflow-status-slack-notification","short":true},{"title":"Branch","value":"main","short":true},{"title":"Action URL","value":"<https://github.com/reside-eng/workflow-status-slack-notification/actions/runs/23456|Publish Action>","short":true},{"title":"Event","value":"pull_request","short":true},{"title":"Publish Action workflow failure","value":"Publish Action workflow in workflow-status-slack-notification failed.","short":false}]}]}',
```

Comment:
Same here, for all these long calledWith assertions.

Reply:
@mathieudi Here is an example of the inline snapshot. It's very useful for reducing maintenance here. Invoke `toMatchInlineSnapshot` without any arguments first, run your tests, and jest will add the expected arguments to your test file. Any change thereafter will be caught. https://github.com/reside-eng/next-bootstrap/blob/main/src/__tests__/withBootstrapConfig.ts#L242-L264